scanner.go

```diff
@@ -294,7 +294,28 @@ func (f *fetcher) update(resp *pb.ScanResponse, region hrpc.RegionInfo) {
 	} else {
 		// we are done with this region, prepare scan for next region
 		f.scannerID = noScannerID
-		f.startRow = region.StopKey()
+
+		// Normal Scan
+		if !f.rpc.Reversed() {
+			f.startRow = region.StopKey()
+			return
+		}
+
+		// Reversed Scan
+		// return if we are at the end
+		if len(region.StartKey()) == 0 {
+			f.startRow = region.StartKey()
+			return
+		}
+
+		// create the nearest value lower than the current region startKey
+		// this won't work correctly on keys  / splits with embedded 0x0.
```

Comment:
Not supporting embedded `0x0` is pretty big limitation IMHO.
Seems like it could be easily solved by doing something like this:
```
if tmp[len(tmp)-1] == 0x0 {
     f.startRow = tmp[:len(tmp)-1]
     return
}
```